otherlibs/unix/access.c

```diff
@@ -24,22 +24,23 @@
 #else
 # ifndef _WIN32
 #  include <sys/file.h>
-#  ifndef R_OK
+# endif
+# ifndef R_OK
 #   define R_OK    4/* test for read permission */
 #   define W_OK    2/* test for write permission */
 #   define X_OK    1/* test for execute (search) permission */
 #   define F_OK    0/* test for presence of file */
-#  endif
-# else
-#  define R_OK    4/* test for read permission */
-#  define W_OK    2/* test for write permission */
-#  define X_OK    4/* test for execute permission - not implemented in Win32 */
-#  define F_OK    0/* test for presence of file */
 # endif
 #endif
 
 static int access_permission_table[] = {
-  R_OK, W_OK, X_OK, F_OK
+  R_OK, W_OK, 
```

Comment:
This is quite difficult to read.

Maybe making intermediate variables like "r_ok", "w_ok, "x_ok" and "f_ok" would be clearer; x_ok's value would depend on some #defines.

Another possibility would also be to split that first line ("R_OK, W_OK,") so that the various lines here get more consistent.

In any case, I think the whole thing is going to be difficult to make pretty but I think it's worth it: the variable value for X_OK was quite surprising.